### DISCUSSION TITLE
Configuration of DKiM module

### DISCUSSION THREAD

**Discussion by chencho80** [rspamd/rspamd#5090] [General]
Hello,

Since the documentation of the DKIM module (https://rspamd.com/doc/modules/dkim.html) is a bit poor, I'll ask here how I can achieve the following goals:

How to accept, reject or tempfail mails with bad DKIM signature?
How to accept, reject or tempfail mails with no DKIM signature?
How to accept, reject or tempfail mails in case of a DNS error?
How to accept, reject or tempfail mails when the key referenced by a signature is not present in the DNS?

Thank you for your help!

**Comment by fatalbanana** [comment]
From a practical point of view I'd suggest not to do it - it's not a useful enough signal to make a decision on. Similar idea was discussed [here](https://github.com/rspamd/rspamd/discussions/4641).

The way to do it is using [force actions module](https://rspamd.com/doc/modules/force_actions.html).

Reasonable alternative might be applying [whitelist module](https://rspamd.com/doc/modules/whitelist.html) to punish DMARC failures for some domains that are expected to pass.
